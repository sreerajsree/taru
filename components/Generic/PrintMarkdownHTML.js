import markdownStyles from '../markdown-styles.module.css';

export default function PrintMarkdownHTML({ content, about }) {
  return (
    <div
      className={about ? markdownStyles.about : markdownStyles.markdown}
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: content }}
    />
  );
}
