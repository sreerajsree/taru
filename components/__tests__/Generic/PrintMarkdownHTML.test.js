import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import PrintMarkdownHTML from '../../Generic/PrintMarkdownHTML';

describe('PrintMarkdownHTML', () => {
  test('it renders and matches snapshot', () => {
    const { asFragment } = render(<PrintMarkdownHTML content="<p>Hello World</p>" />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Hello World')).toBeInTheDocument();
  });
});
