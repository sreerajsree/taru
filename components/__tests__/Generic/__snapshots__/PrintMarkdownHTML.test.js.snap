// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintMarkdownHTML it renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="markdown"
  >
    <p>
      Hello World
    </p>
  </div>
</DocumentFragment>
`;
